app/utils/draft/index.ts

```diff
@@ -33,3 +47,242 @@ export function alertErrorWithFallback(intl: IntlShape, error: any, fallback: Me
     }
     Alert.alert('', msg, buttons);
 }
+
+export function alertAttachmentFail(intl: IntlShape, accept: AlertCallback, cancel: AlertCallback) {
+    Alert.alert(
+        intl.formatMessage({
+            id: 'mobile.post_textbox.uploadFailedTitle',
+            defaultMessage: 'Attachment failure',
+        }),
+        intl.formatMessage({
+            id: 'mobile.post_textbox.uploadFailedDesc',
+            defaultMessage: 'Some attachments failed to upload to the server. Are you sure you want to post the message?',
+        }),
+        [{
+            text: intl.formatMessage({id: 'mobile.channel_info.alertNo', defaultMessage: 'No'}),
+            onPress: cancel,
+        }, {
+            text: intl.formatMessage({id: 'mobile.channel_info.alertYes', defaultMessage: 'Yes'}),
+            onPress: accept,
+        }],
+    );
+}
+
+export const textContainsAtAllAtChannel = (text: string) => {
+    const textWithoutCode = text.replace(CODE_REGEX, '');
+    return (/(?:\B|\b_+)@(channel|all)(?!(\.|-|_)*[^\W_])/i).test(textWithoutCode);
+};
+
+export const textContainsAtHere = (text: string) => {
+    const textWithoutCode = text.replace(CODE_REGEX, '');
+    return (/(?:\B|\b_+)@(here)(?!(\.|-|_)*[^\W_])/i).test(textWithoutCode);
+};
+
+export const groupsMentionedInText = (groupsWithAllowReference: Group[], text: string) => {
+    if (!groupsWithAllowReference.length) {
+        return [];
+    }
+    const textWithoutCode = text.replace(CODE_REGEX, '');
+    const mentions = textWithoutCode.match(AT_MENTION_REGEX_GLOBAL) || [];
+    return groupsWithAllowReference.filter((g) => mentions.includes(g.id));
+};
+
+export const mapGroupMentions = (channelMemberCountsByGroup: ChannelMemberCountByGroup[], groupMentions: Group[]) => {
```

Comment:
This has been brought more or less as is from V1. The name is not revealing enough. What it does is:
- For each group mention, add the name to a set (to avoid duplicates, I guess)
- If any group has more members than the limit on `NOTIFY_ALL_MEMBERS`, return also the number of members to notify and the timezones count.

These values are used to check if the mention should trigger the alert, and fill the information on the alert.

Not sure if we should change the name of the function and how exactly explain this in the comment.

Reply:
Thanks for the explanation,  instead of `description`, I meant ` comment` - function comment